core/frontend/src/utils/pattern_validators.ts

```diff
@@ -0,0 +1,33 @@
+export function validate_port(port: number, public_range = true): boolean {
+  if (!Number.isInteger(port)) {
+    return false
+  }
+  if (port > 65535) {
+    return false
+  }
+  if (public_range) {
+    return port >= 1024
+  }
+  return port >= 0
+}
+
+export function validate_baudrate(baudrate: number): boolean {
+  const VALID_SERIAL_BAUDRATES = [
+    3000000, 2000000, 1000000, 921600, 570600, 460800, 257600, 250000, 230400, 115200, 57600, 38400, 19200, 9600,
+  ]
+  return VALID_SERIAL_BAUDRATES.includes(baudrate)
+}
+
+export function validate_ip(ip: string): boolean {
+  const ip_pattern = /^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/
+  return ip_pattern.test(ip)
+}
+
+export function validate_filepath(filepath: string): boolean {
+  const filepath_pattern = /^(.+)\/([^/]+)$/
+  return filepath_pattern.test(filepath)
+}
+
+export function validate_non_empty(input: string): boolean {
```

Comment:
I believe that is better to make it generic to work for any scenario.
```ts
function is_empty<T>(sizable: Array<T> | string | undefined | null) {
    return !sizable || sizable.length == 0
}
```

Reply:
Cool! Will update.